### PR TITLE
Make courtCase in PNC details panel collapsible 

### PR DESCRIPTION
**Pull request by tausifpatel**
Expanded
Screenshot 2024-07-19 at 14 49 13

The first accordion is open by default. Additional Accordions are collapsed by default.

Collapsed
Screenshot 2024-07-19 at 14 49 38

On Hover
Screenshot 2024-07-19 at 14 53 18

### CODE REVIEW

File: cypress/component/PncDetails.cy.tsx [hunk 1]
```diff
@@ -189,4 +189,123 @@ describe("PNC details", () => {
...
+
+    cy.mount(
+      <CurrentUserContext.Provider value={{ currentUser }}>
+        <CourtCaseContext.Provider value={[{ courtCase, amendments: {}, savedAmendments: {} }, () => {}]}>
+          <PncDetails />
+        </CourtCaseContext.Provider>
+      </CurrentUserContext.Provider>
+    )
+
+    cy.get(".pnc-offence").first().should("exist")
```
Comment (richrace) [inline cypress/component/PncDetails.cy.tsx:308]: Use consistent expectations

```suggestion
    cy.get(".pnc-offence").eq(0).should("exist")
```

File: src/features/CourtCaseDetails/Sidebar/PncDetails/PncCourtCaseAccordion.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+const PncCourtCaseAccordion = ({
+  index,
+  pncCourtCase: { courtCaseReference, crimeOffenceReference, offences }
+}: PncCourtCaseAccordionProps) => {
+  const [open, setOpen] = useState<boolean>(index === 0 ? true : false)
+  const handleClick = () => {
+    setOpen(!open)
+  }
```
Comment (richrace) [inline src/features/CourtCaseDetails/Sidebar/PncDetails/PncCourtCaseAccordion.tsx:27]:
```suggestion
  const handleClick = () => setOpen((previousState) => !previousState)

```

File: src/features/CourtCaseDetails/Sidebar/PncDetails/PncCourtCaseAccordion.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          <CrimeOffenceReference>
+            <div className={"heading"}>{"Crime Offence Reference"}</div>
+            <div id={"crime-offence-reference"}>{crimeOffenceReference || "-"}</div>
+          </CrimeOffenceReference>
+        </CourtCaseHeader>
+
+        <ChevronContainer>
+          <span className={`govuk-accordion-nav__chevron ${chevronPosition} chevron`}></span>
+        </ChevronContainer>
+      </CourtCaseHeaderContainer>
```
Comment (richrace) [inline src/features/CourtCaseDetails/Sidebar/PncDetails/PncCourtCaseAccordion.tsx:45]: Looking at `src/components/Accordion.tsx`, we should have some `aria` attributes for accessibility.

File: src/features/CourtCaseDetails/Sidebar/PncDetails/PncDetails.styles.tsx [hunk 4]
```diff
@@ -22,23 +22,48 @@ const CourtCase = styled.div`
...
+  justify-content: flex-end;
+  flex-grow: inherit;
+  align-items: center;
+  margin-right: 15px;
+`
+
+const CourtCaseHeader = styled.div`
+  width: 100%;
+
+  margin: 15px 20px;
```
Comment (richrace) [inline src/features/CourtCaseDetails/Sidebar/PncDetails/PncDetails.styles.tsx:53]: Can we please create a separate file for the `PncCourtCaseAccordion` styled components?

File: src/features/CourtCaseDetails/Sidebar/PncDetails/PncCourtCaseAccordion.tsx (hunk 2 above)
Comment (richrace) [reply src/features/CourtCaseDetails/Sidebar/PncDetails/PncCourtCaseAccordion.tsx:27]: Also, why did we remove the `toggle` word? In `src/components/Accordion.tsx` it's used. I'd like to keep things consistent with the Accordion.

**Comment by richrace** [comment]
I understand why we can't use the Accordion component, but we should stay as close to it as possible.

File: src/features/CourtCaseDetails/Sidebar/PncDetails/PncDetails.styles.tsx (hunk 4 above)
Comment (tausifpatel) [reply src/features/CourtCaseDetails/Sidebar/PncDetails/PncDetails.styles.tsx]: Created a separate styled component

**Review by richrace — APPROVED** [review]

**Pull request merged**